core/src/main/scala/caliban/schema/Schema.scala

```diff
@@ -105,7 +105,7 @@ trait Schema[-R, T] { self =>
       self.toType_(isInput, isSubscription).copy(name = Some(newName))
     }
 
-    private val renameTypename: Boolean = self.toType_().kind match {
+    private lazy val renameTypename: Boolean = self.toType_().kind match {
```

Comment:
Can you add the regression test for this?

Reply:
The test that I added was the repro for this fix. It just happens that the same test would also fail for the Scala 3 union issue, so I reused it

Reply:
Aha, okay. I just saw one so I thought it didn't cover both.